### PR TITLE
Bump x509 parser

### PR DESCRIPTION
**Pull request by cijiugechu**
`x509-parser 0.9` depends on `nom 0.6`, which causes future warnings: 
```
warning: the following packages contain code that will be rejected by a future version of Rust: nom v6.1.2
```
in [gleam lang repo](https://github.com/gleam-lang/gleam), so updgrade that version.

### CODE REVIEW

**Review by lpil — COMMENTED** [review]
Thank you.

What are the other changes for? They seem unrelated.

File: src/lib.rs [hunk 1]
```diff
@@ -286,6 +287,7 @@ pub fn get_package_response(
     match parts.status {
         StatusCode::OK => (),
         StatusCode::FORBIDDEN => return Err(ApiError::NotFound), // Oddly this is the not-found code
+        StatusCode::NOT_FOUND => return Err(ApiError::NotFound),
```
Comment (cijiugechu) [inline src/lib.rs:290]: This line is actually unrelated to the dependency upgrade. It is changed because now the `get_package_response` and `get_package_tarball_response` functions return `UnexpectedResponse` instead of `NotFound`, I modified these two functions. Should I split the current PR into two?

Comment (lpil) [reply src/lib.rs:290]: Yes please, another PR. We'll need tests for it too

**Review by lpil — APPROVED** [review]
Thank you!
